docs/adrs/0078-develop-strategy-for-ensuring-code-test-coverage.md

```diff
@@ -0,0 +1,61 @@
+---
+title: 0078 Develop strategy for ensuring code test coverage
+description: |
+  Decision outcome: The Happo Approach
+---
+
+# Develop strategy for ensuring code test coverage
+
+**User Story:** *[MB-16176](https://dp3.atlassian.net/browse/MB-16176)* :lock:
+
+## Problem Statement
+
+Our current system of maintaining test coverage involves measuring the current total test coverage and ensuring that any given PR does not decrease the total coverage.
+This strategy has obvious benefits in helping to support a high test coverage, but it can also introduce unnecessary roadblocks in the PR workflow. To understand this, it's helpful to understand the scenarios that might result in a decrease in total coverage.
+
+
+### Scenarios that result in a test coverage decrease
+1. **Removal of code that is _more_ tested than the total coverage.** If the total coverage is 77.0% and a block of code that is tested at 100% is deleted, the average coverage goes down.
+2. **Addition of code that is _less_ tested than the total coverage and _should be tested more_.** This is the scenario that we want to address with our code coverage strategy.
+3. **Addition of code that is _less_ tested than the total coverage and _doesn't need to be tested more_.** The difference between this scenario and scenario 2 can be subjective, but one can imagine a scenario in which a Go database model is created and doesn't require any tests.
```

Comment:
Are these the only scenarios we'd like to discuss here? I think these are enough, but I'd like to hear if folks can think of another or perhaps expand on these scenarios a bit more with more examples.

Reply:
**thought**: I think another example for scenario 3 would be adding logic to existing model_to_payload functions since those aren't unit tested and are only sometimes integration tested via the handler integration tests.